### PR TITLE
database/raft: avoid server cert errors for peers

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

File: database/raft/raft.go [hunk 1]
```diff
@@ -147,6 +147,11 @@ type Getter interface {
 func Start(laddr, dir, bootURL string, httpClient *http.Client) (*Service, error) {
```
Comment (tessr) [inline database/raft/raft.go]: I know this was a previous PR, but while we're poking at this, can you add `httpClient` to the doc comment for `Start`?

File: database/raft/tls.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+	"chain/errors"
+)
+
+func verifyTLSName(name string, client *http.Client) error {
+	c := clientTLS(client)
+	if c == nil {
+		return nil
+	}
+	x509Cert, err := x509.ParseCertificate(c.Certificates[0].Certificate[0])
```
Comment (tessr) [inline database/raft/tls.go]: Is it possible that this line will cause an index out of bounds? (And if so, is crashing the desired behavior here?)

**Comment by jbowens** [comment]
Is this a raft-specific check or does it apply more generally to cored too?

File: database/raft/tls.go (hunk 2 above)
Comment (kr) [inline database/raft/tls.go]: If there's a TLS client config at all here, it should definitely have a cert, so I think crashing is fine.

**Comment by kr** [comment]
Yeah it is more general. It applies anywhere we have an address or hostname we are planning to tell other processes to use to reach us. So raft and the leader key. We could add another check for the leader key or we could potentially have the leader stuff piggyback on raft's concept of identity. (Like, use the raft cluster member id as the leader key and also have a way to look up an address from a member id. idk)

**Comment by kr** [comment]
Or maybe we should just move this check out into package main. Not sure.

**Pull request merged**